controllers/httproute_controller.go

```diff
@@ -336,11 +339,41 @@ func (r *HTTPRouteReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	gwEventHandler := eventhandlers.NewEnqueueRequestGatewayEvent(r.Client)
 	svcEventHandler := eventhandlers.NewEqueueHTTPRequestServiceEvent(r.Client)
 	svcImportEventHandler := eventhandlers.NewEqueueRequestServiceImportEvent(r.Client)
-	return ctrl.NewControllerManagedBy(mgr).
+
+	builder := ctrl.NewControllerManagedBy(mgr).
 		// Uncomment the following line adding a pointer to an instance of the controlled resource as an argument
 		For(&gateway_api.HTTPRoute{}).
 		Watches(&source.Kind{Type: &gateway_api.Gateway{}}, gwEventHandler).
 		Watches(&source.Kind{Type: &corev1.Service{}}, svcEventHandler).
-		Watches(&source.Kind{Type: &mcs_api.ServiceImport{}}, svcImportEventHandler).
-		Complete(r)
+		Watches(&source.Kind{Type: &mcs_api.ServiceImport{}}, svcImportEventHandler)
+
+	if ok, err := isExternalDNSSupported(mgr); ok {
+		builder.Owns(&endpoint.DNSEndpoint{})
+	} else if err != nil {
+		glog.V(2).Infof("Unknown error while discovering CRD: %v", err)
```

Comment:
This is concerning.  Looks like if ExternalDNS is not supported, the controller will `not` run.

Reply:
`err != nil` just means that it failed to run discovery API for some reason. If ExternalDNS isn't supported the result is `(false, nil)` so it gracefully skips the logic.

Reply:
can u add this to the comment? it was not obvious to me when I read this code. thanks

Reply:
fixed